.github/workflows/publish-client.yml

```diff
@@ -0,0 +1,165 @@
+---
+# Automatically build and publish the client to the container registry.
+# This workflow is triggered on changes to the client directory.
+
+name: Build and Publish hmi-client
+
+# yamllint disable-line rule:truthy
+on:
+  push:
+    paths: [ 'packages/client/**' ]
+    branches: [ 'main' ]
+    tags: [ '*' ]
+
+jobs:
+  image-name:
+    name: Create image name
+    runs-on: ubuntu-22.04
+    outputs:
+      name: ${{ steps.define.outputs.name }}
+    steps:
+      - id: define
+        run: |
+          if [[ '${{ github.ref_type }}' == 'branch' && '${{ github.ref_name }}' == 'main' ]]; then
+            TAG=latest
+          else
+            SEMVER=$( echo ${{ github.ref_name }} | sed -nre 's/^v[^0-9]*(([0-9]+\.)*[0-9]+(-[a-z]+)?).*/\1/p')
+            if [[ -n $SEMVER ]]; then
+              TAG=${SEMVER}
+            else
+              TAG=${{ github.ref_name }}
+            fi
+          fi
+
+          IMAGE=ghcr.io/darpa-askem/hmi-client:${TAG,,}
+          echo "$IMAGE"
+          echo "name=$IMAGE" >> $GITHUB_OUTPUT
+
+  amd64:
+    runs-on: ubuntu-22.04
+    needs: image-name
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Login to registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Setup Node 20.10.0 LTS
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.10.0
+
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
+      - name: Build image using Makefile
```

Comment:
Why are you building here and then just below with the action? choose one method

Reply:
It's just a poorly formatted name from the makefile as it doesn't build the image, but build the hmi-client (from source to distribution)